holoviews/plotting/bokeh/chart.py

```diff
@@ -906,25 +903,39 @@ def get_data(self, element, ranges, style):
         hover = 'hover' in self.handles
 
         # Group by stack or group dim if necessary
+        xdiff = None
+        xvals = element.dimension_values(xdim)
         if group_dim is None:
             grouped = {0: element}
+            is_dt = isdatetime(xvals)
+            if is_dt or xvals.dtype.kind != 'O':
```

Comment:
Should exclude fixed length unicode

```suggestion
            if is_dt or xvals.dtype.kind not in 'OU':
```